### PR TITLE
feat: update el requests for devnet 4

### PR DESCRIPTION
**Pull request by onbjerg**
Bumps reth and alloy, and makes changes to how requests are handled throughout reth (both in syncing and in building) in preparation for devnet 4. Specifically:

- Requests are now opaque and can be ferried around as-is in the representation their respective system contracts return them
- We no longer need to serialize or deserialize requests, ever
- We only need to perform 1 check for request validity: that `requests_hash` in the header matches `requests_hash` that we computed
- `requests_root` has been replaced with a `requests_hash`

This also removes the `Requests` table, and removes requests from P2P, as it doesn't make sense to gossip requests, and I think we're the only ones doing it (oops). Wrt removing `Requests`, there are still no RPC endpoints that expose them, so I think we should remove it as it was prematurely added. It does not make sense to expose the requests over RPC anyway, as they are now opaque types only meant for the CL.

See 

- https://github.com/ethereum/execution-apis/pull/591
- https://github.com/ethereum/EIPs/pull/8949 (and related PRs)

Depends on 
- https://github.com/alloy-rs/alloy/pull/1383
- https://github.com/alloy-rs/op-alloy/pull/176
- https://github.com/paradigmxyz/reth/pull/11870

### CODE REVIEW

**Comment by onbjerg** [comment]
This also requires a lot of changes in op-alloy it seems. Seems like there were a lot of breaking changes in this alloy bump :)

File: crates/rpc/rpc-api/src/engine.rs [hunk 1]
```diff
@@ -137,7 +138,7 @@ pub trait EngineApi<Engine: EngineTypes> {
     /// payload build process at the time of receiving this call. Note:
     /// > Provider software MAY stop the corresponding build process after serving this call.
     #[method(name = "getPayloadV4")]
-    async fn get_payload_v4(&self, payload_id: PayloadId) -> RpcResult<Engine::ExecutionPayloadV4>;
+    async fn get_payload_v4(&self, payload_id: PayloadId) -> RpcResult<Engine::ExecutionPayloadV3>;
```
Comment (mattsse) [inline crates/rpc/rpc-api/src/engine.rs:141]: this must be V4 right?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/ethereum/engine-primitives/src/payload.rs [hunk 2]
```diff
@@ -142,10 +141,17 @@ impl From<EthBuiltPayload> for ExecutionPayloadEnvelopeV3 {
...
-
+        let EthBuiltPayload { block, fees, sidecars, executed_block, .. } = value;
+
+        // if we have an executed block, we pop off the first set of requests from the execution
+        // outcome. the assumption here is that there will always only be one block in the execution
+        // outcome.
+        let execution_requests = executed_block
+            .and_then(|block| block.execution_outcome().requests.first().cloned())
+            .map(Requests::take)
+            .unwrap_or_default();
```
Comment (mattsse) [inline crates/ethereum/engine-primitives/src/payload.rs:152]: this isn't ideal but works for vanilla payload builder, can improve later

**Review by mattsse — APPROVED** [review]

**Comment by onbjerg** [comment]
Not going to use https://github.com/alloy-rs/alloy/pull/1517 in this PR, but will create a follow up issue and work on it after to clean it up a bit.

File: bin/reth/src/commands/debug_cmd/replay_engine.rs [hunk 3]
```diff
@@ -171,7 +171,9 @@ impl<C: ChainSpecParser<ChainSpec = ChainSpec>> Command<C> {
                     debug!(target: "reth::cli", ?response, "Received for forkchoice updated");
                 }
                 StoredEngineApiMessage::NewPayload { payload, cancun_fields } => {
-                    let response = beacon_engine_handle.new_payload(payload, cancun_fields).await?;
+                    // todo: prague (last arg)
+                    let response =
+                        beacon_engine_handle.new_payload(payload, cancun_fields, None).await?;
```
Comment (mattsse) [inline bin/reth/src/commands/debug_cmd/replay_engine.rs:176]: will take care of in followup

**Review by mattsse — APPROVED** [review]

**Pull request merged**